src/xrEngine/xr_ioc_cmd.cpp

```diff
@@ -536,18 +536,33 @@ virtual void Save (IWriter *F) {};
 
 ENGINE_API BOOL r2_sun_static = TRUE;
 ENGINE_API BOOL r2_advanced_pp = FALSE; // advanced post process and effects
+ENGINE_API BOOL renderer_allow_override = FALSE;
 
-u32 renderer_value = 3;
-
-class CCC_r2 : public CCC_Token
+class CCC_renderer : public CCC_Token
 {
     typedef CCC_Token inherited;
 
+    u32 renderer_value = 3;
+    static BOOL cmd_lock;
```

Comment:
No, please, remove this lock. It breaks the ability to change the renderer more than once.

Reply:
Actually no. The `cmd_lock` has effect only until `renderer_allow_override` set to `FALSE` (default value). If no `-rXX` cmd line options provided the `renderer` command will behave as usually. Please refer to added line 295 in `main.cpp` and statement at line 553 in `xr_ioc_cmd.cpp`.

In other words there are two possible scenarios:
1. Application has `-rXX` set and use it as only option (further renderer changes won't have effect)
2. No renderer overridings provided = regular case. The command works as is.

Reply:
Ah! Then it's okay.

But, then we should lock the renderer dropbox in the options, so that user will know that he can't change it because cmd key is set.

Can you try to do this? It it's impossible with the current scripting abilities, then we should open an issue for that)